uint64 id_initial
uint64 id_current
